test/integration/puppetlabs/services/legacy_routes/legacy_routes_test.clj

```diff
@@ -25,39 +28,126 @@
+       (is (nil? (schema/check jruby-metrics-core/JRubyMetricsStatusV1
+                               (get-in status [:jruby-metrics :status]))))
+
+       (is (= jruby-metrics-core/jruby-pool-lock-not-in-use
+              (get-in status [:jruby-metrics :status :experimental
+                              :jruby-pool-lock-status :current-state])))
+
+       (is (= 1 (get-in status [:master :service_status_version])))
+       (is (= "running" (get-in status [:master :state])))
+       (is (nil? (schema/check master-core/MasterStatusV1
+                               (get-in status [:master :status]))))
+       (testing (str "HTTP metrics in status endpoint are sorted in order of "
+                     "aggregate amount of time spent")
+         (let [hit-routes #{"total"
+                            "puppet-v3-environments"
+                            "puppet-v3-node-/*/"
+                            "puppet-v3-catalog-/*/"
+                            "other"}
+               http-metrics (get-in status [:master
+                                            :status
+                                            :experimental
+                                            :http-metrics])]
+           (testing (str "'total' should come first since it is the sum of "
+                         "the other endpoints")
+             (is (= "total" (:route-id (first http-metrics)))))
+           (testing (str "The other three routes that actually received "
+                         "requests should come next")
+             (is (= #{"puppet-v3-environments"
+                      "puppet-v3-node-/*/"
+                      "puppet-v3-catalog-/*/"}
+                    (set (map :route-id (rest (take 4 http-metrics)))))))
+           (testing "The aggregate times should be in descending order"
+             (let [aggregate-times (map :aggregate http-metrics)]
+               (= aggregate-times (reverse (sort aggregate-times)))))
+           (testing (str "The counts should be accurate for the endpoints "
+                         "that we hit")
+             (let [find-route (fn [route-metrics route-id]
+                                (first (filter #(= (:route-id %) route-id) route-metrics)))]
+               (is (= 4 (:count (find-route http-metrics "total"))))
+               (is (= 1 (:count (find-route http-metrics "puppet-v3-environments"))))
+               (is (= 1 (:count (find-route http-metrics "puppet-v3-node-/*/"))))
+               (is (= 1 (:count (find-route http-metrics "puppet-v3-catalog-/*/"))))
+               (is (= 1 (:count (find-route http-metrics "other"))))))
+           (testing "The counts should be zero for endpoints that we didn't hit"
+             (is (every? #(= 0 %) (map :count
+                                       (filter
+                                        #(not (hit-routes (:route-id %)))
+                                        http-metrics)))))))))
```

Comment:
So this is now like a 200 line test... it seems like it would be good to break it up a bit, and this seems like a reasonable point to do it. Have a test for the metrics in `master -> experimental -> http-metrics` and then have a test that these request show up in the `jruby-metrics` when they have borrowed a jruby.